core/types/src/macros.rs

```diff
@@ -100,12 +121,27 @@ macro_rules! impl_newtype_for_bytestruct {
             }
         }
 
+        impl core::fmt::Display for $wrapper {
+            fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+                write!(f, "0x")?;
+                let inner: &[u8] = self.as_ref();
+                for byte in inner {
+                    write!(f, "{:02X}", byte)?;
+                }
+                Ok(())
```

Comment:
It looks like https://doc.rust-lang.org/std/fmt/trait.UpperHex.html isn't implemented for &[u8] and it seemed a little heavy to bring in the hex crate for only this.

Reply:
added underscores.
Also moved this to an `UpperHex` impl and have `Display` redirect to that.